Sources/Build/BuildDelegate.swift

```diff
@@ -315,4 +328,147 @@ public final class BuildDelegate: BuildSystemDelegate {
     public func shouldResolveCycle(rules: [BuildKey], candidate: BuildKey, action: CycleAction) -> Bool {
         return false
     }
+
+    func swiftCompilerDidOutputMessage(_ message: SwiftCompilerMessage) {
+        queue.sync {
+            taskTracker.swiftCompilerDidOuputMessage(message)
+            updateProgress()
+
+            if let output = message.standardOutput {
+                progressAnimation.clear()
+                outputStream <<< output
+                outputStream.flush()
+            }
+        }
+    }
+
+    func swiftCompilerOutputParserDidFail(withError error: Error) {
+        let message = (error as? LocalizedError)?.errorDescription ?? error.localizedDescription
+        diagnostics.emit(data: SwiftCompilerOutputParsingError(message: message))
+        onCommmandFailure?()
+    }
+
+    private func updateProgress() {
+        if let progressText = taskTracker.oldestRunningText {
+            progressAnimation.update(
+                step: taskTracker.finishedCount,
+                total: taskTracker.totalCount,
+                text: progressText)
+        }
+    }
+}
+
+// Tracks tasks based on command status and swift compiler output.
+fileprivate struct TaskTracker {
```

Comment:
Maybe rename to CompilerTaskTracker?

Reply:
This is for all "tasks" (llbuild commands + compiler tasks), so `CompilerTaskTracker` might be misleading. Any ideas?

Reply:
Do you have other suggestions? TaskTracker seems very vague. Maybe CommandTaskTracker or something?

Reply:
(I've not looked at the code very closely to have a good suggestion)

Reply:
CommandTaskTracker sounds like a good name